# RAG evaluation

Run the current end-to-end hybrid RAG quality evaluation against the 105-case golden dataset. It evaluates the configured RAG pipeline without the optional LLM judge and does not compare prompts or models.

This makes one generated answer per golden case and reports retrieval, citation, refusal, and latency metrics for 100 answerable and 5 unsupported questions. Ensure the agent OpenAI settings are available in the environment or `.env` file.

In [1]:
from pathlib import Path
from subprocess import run

project_root = Path.cwd().resolve()
if not (project_root / 'pyproject.toml').is_file():
    project_root = project_root.parents[1]

run(['make', 'evaluate-rag'], cwd=project_root, check=True)

{
  "dataset": "evaluation/datasets",
  "corpus": "fixtures/corpus",
  "mode": "hybrid",
  "k": 5,
  "metrics": {
    "cases": 105,
    "source_precision": 0.9810,
    "source_recall": 0.9810,
    "section_precision": 0.9008,
    "section_recall": 0.9619,
    "citation_validity": 0.9857,
    "citation_coverage": 0.9810,
    "refusal_rate": 0.0,
    "refusal_accuracy": 0.9524,
    "error_rate": 0.0,
    "mean_retrieval_latency_ms": 46.03,
    "mean_generation_latency_ms": 15329.62,
    "mean_total_latency_ms": 15375.90
  }
}


**Conclusion:** The current hybrid RAG pipeline has strong document-level quality across 105 cases, with 98.1% source precision/recall, 96.2% section recall, 98.6% citation validity, and no errors. Section precision is 90.1%, so some answers still cite extra or imprecise sections. The five unsupported cases exposed a refusal gap: refusal rate is 0% and refusal accuracy is 95.2%, meaning the current prompt did not refuse those cases. Retrieval remains fast at 46 ms; generation dominates latency at about 15.3 seconds per case before asynchronous overlap.